parmed/charmm/parameters.py

```diff
@@ -430,8 +430,17 @@ def read_parameter_file(self, pfile, comments=None):
                     raise CharmmError('Could not parse bonds.')
                 key = (min(type1, type2), max(type1, type2))
                 bond_type = BondType(k, req)
-                self.bond_types[(type1, type2)] = bond_type
-                self.bond_types[(type2, type1)] = bond_type
+                if key in self.bond_types:
+                    # See if the existing bond type list has a different value and replaces it with a warning
+                    if self.bond_types[key] != bond_type:
+                        # Replace. Warn if they are differen
+                        warnings.warn('Replacing bond %r, %r with %r' %
+                                              (key, self.bond_types[key], bond_type))
```

Comment:
Warn with a `ParameterWarning` (which, if it's not already imported, is found in `parmed.exceptions`)